cns/client/client.go

```diff
@@ -1050,7 +1058,11 @@ func (c *Client) GetEndpoint(ctx context.Context, endpointID string) (*restserve
 		return nil, errors.Wrap(err, "failed to decode GetEndpointResponse")
 	}
 
+	if response.Response.ReturnCode == types.NotFound {
```

Comment:
did we make sure `types.NotFound` set by CNS when endpoint not found in its state?

Reply:
yes, we were already doing it on API handler in CNS for that specific error